### PR TITLE
feat: add connector unique name suffix

### LINKED ISSUE #2793: [BUG] When adding a new Alby account, shouldn't be able to add an account with same name of an existing account
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

Hi,

When adding a new Alby account, shouldn't be able to add an account with same name of an existing account. Should ask to add a new name if already one with same name exists.

### Screenshots [optional]

_No response_

### Steps To Reproduce

1- Add a new Alby custodial wallet

### Expected behavior

Ask to rename the new account if a existing one exist with the same name.
Why? It's very confusing have two accounts with the same name. 

### Alby information

Alby: 3.2.1

### Device information

_No response_

### Additional context

_No response_

### Are you working on this?

No

### PR DESCRIPTION
**Pull request by lujakob**
### Describe the changes you have made in this PR

The accounts in the list of accounts under "Manage accounts" should have unique names. The changes of this PR add a "(1)" suffix to the account name in case an account of the same name already exists. The number in between the brackets "()" is detected and increased by 1, so an account with the same name can be added multiple times and the suffix will be increased from "(1)" to "(2)" to "(3)" etc...

### Link this PR to an issue [optional]

Fixes #2793

### Type of change

- `feat`: New feature (non-breaking change which adds functionality)

### Screenshots of the changes [optional]

Bildschirmfoto 2023-11-14 um 21 58 13

### How has this been tested?

Manually

### Checklist

- [x] Self-review of changed code
- [x] Manual testing
- [x] Added automated tests where applicable
- [x] Update Docs & Guides
- For UI-related changes
- [x] Darkmode
- [x] Responsive layout

### CODE REVIEW

File: src/extension/background-script/actions/accounts/add.ts [hunk 1]
```diff
@@ -1,13 +1,33 @@
...
+import type { Account, MessageAccountAdd } from "~/types";
 
 const add = async (message: MessageAccountAdd) => {
   const newAccount = message.args;
+  let name = newAccount.name;
   const accounts = state.getState().accounts;
   const tmpAccounts = { ...accounts };
 
+  // get all accounts names to avoid duplicate names
+  const accountNames = Object.values(accounts).map((el: Account) => el.name);
```
Comment (reneaaron) [inline src/extension/background-script/actions/accounts/add.ts:13]: Quite hard to understand what is happening here exactly. 🤔

Can we extract a method for this logic and add some unit tests? I'll try to wrap my head around it in the meantime 😅

**Review by reneaaron — COMMENTED** [review]
Thanks for the PR @lujakob, great to see you around again! 😊

File: src/extension/background-script/actions/accounts/add.ts (hunk 1 above)
Comment (reneaaron) [reply src/extension/background-script/actions/accounts/add.ts:13]: Also prettier seems to fail for this file, you might want to have a look at that.

**Comment by lujakob** [comment]
> @reneaaron I added tests as suggested and ran the format command. Please have another look.

Hi @reneaaron could you please check this PR once again? Thanks.

**Review by reneaaron — APPROVED** [review]
Thanks for the PR @lujakob! :raised_hands: 

I tried to simplify the unique name generation a bit, have a look if that makes sense!

tACK

**Comment by lujakob** [comment]
> Thanks for the PR @lujakob! 🙌
> 
> I tried to simplify the unique name generation a bit, have a look if that makes sense!
> 
> tACK

Awesome, thanks.

**Pull request merged**